posydon/binary_evol/DT/magnetic_braking.py

```diff
@@ -0,0 +1,361 @@
+        stars, respectively (units of yr^-2)
+
+    """
+    if magnetic_braking_mode == "RVJ83":
+        return calculate_magnetic_braking_RVJ83(p_pri, p_sec)
+    elif magnetic_braking_mode == "M15":
+        return calculate_magnetic_braking_M15(p_pri, p_sec)
+    elif magnetic_braking_mode == "G18":
+        return calculate_magnetic_braking_G18(p_pri, p_sec)
+    elif magnetic_braking_mode == "CARB":
+        return calculate_magnetic_braking_CARB(p_pri, p_sec)
+    else:
+        raise Exception("WARNING: Magnetic braking is not being calculated in"
+                        "the detached step. The given magnetic_braking_mode"
+                        "string \"", magnetic_braking_mode, "\" does not match"
+                        "the available built-in cases. To enable magnetic"
+                        "braking, please set magnetic_braking_mode to one of"
+                        "the following strings:\n"
+                        "\"RVJ83\" for Rappaport, Verbunt, & Joss 1983\n"
+                        "\"G18\" for Garraffo et al. 2018\n"
+                        "\"M15\" for Matt et al. 2015\n"
+                        "\"CARB\" for Van & Ivanova 2019\n")
+
+
+def calculate_magnetic_braking_RVJ83(p_pri, p_sec):
+    """Use the Rappaport, Verbunt, and Joss (1983) prescription.
+
+    Calculate the impact of magnetic braking on a binary using the
+    Rappaport, Verbunt, and Joss 1983, ApJ, 275, 713 prescription. The torque
+    is taken from eq.36 of Rapport+1983, with γ = 4. Torque units
+    # converted from cgs units to [Msol], [Rsol], [yr] as all stellar
+    # parameters are given in units of [Msol], [Rsol], [yr] and so that
+    # dOmega_mb/dt is in units of [yr^-2].
+
+    Parameters
+    ----------
+    p_pri, p_sec : two tuples
+        The properties of the primary and secondary stars, respectively. Each
+        tuple contains, in order, M_pri (the star's mass in Msun), R_pri
+        (the star's radius in Rsun), Omega_pri (the star's angular velocity in
+        1/yr), I_pri (the star's moment of inertia in Msun Rsun^2),
+        tau_conv_pri (the convective timescale of the primary),
+        and Mdot_pri (the mass loss rate of the star in Msun/yr)
+
+    Returns
+    -------
+    dOmega_mb_sec, dOmega_mb_pri : float
+        The spin angular velocity derivative for the secondary and primary
+        stars, respectively [units of yr^-2]
+
+    """
+    M_pri, R_pri, Omega_pri, I_pri, tau_conv_pri, Mdot_pri = p_pri
+    M_sec, R_sec, Omega_sec, I_sec, tau_conv_sec, Mdot_sec = p_sec
+
+    # Converting units:
+    # The constant 3.8e-30 from Rappaport+1983 has units of [cm^-2 s]
+    # which need to be converted...
+    #
+    # -3.8e-30 [cm^-2 s] * (const.rsol**2/const.secyer) -> [Rsol^-2 yr]
+    # * M [Msol]
+    # * R ** 4 [Rsol^4]
+    # * Omega ** 3 [yr^-3]
+    # / I [Msol Rsol^2 ]
+    #
+    # Thus, dOmega/dt comes out to [yr^-2]
+
+    dOmega_mb_sec = (
+        -3.8e30 * (const.rsol**2 / const.secyer)
```

Comment:
My mistake, here's a typo: this should be -3.8e-30 (as in Rappaport, Verbunt, and Joss) not -3.8e30